### PR TITLE
Added Navbar component

### PR DESCRIPTION
**Pull request by mikefrancis**
screen shot 2017-12-05 at 11 36 33
screen shot 2017-12-05 at 11 36 38

### CODE REVIEW

File: src/assets/img/icons/general/chevron-down.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg"><path d="M1683 808l-742 741q-19 19-45 19t-45-19L109 808q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"/></svg>
```
Comment (matine) [inline src/assets/img/icons/general/chevron-down.svg]: The chevrons need to be added in src/assets/img/css-bg/icons

File: src/assets/scss/1-settings/_settings.navbar.scss [hunk 2]
```diff
@@ -0,0 +1,2 @@
+$dropdown-background-colour: #fff;
+$caret-size: 10px;
```
Comment (matine) [inline src/assets/scss/1-settings/_settings.navbar.scss]: You could get these from $white and $icon-size-xxs

File: src/assets/scss/5-objects/_objects.nav.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.nav {
```
Comment (matine) [inline src/assets/scss/5-objects/_objects.nav.scss]: If you wanted to remove all list styling from list elements you can use the mixin: @include remove-list-styles;

File: src/assets/scss/5-objects/_objects.nav.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    padding: 0;
+}
+
+.nav__item {
+    display: inline-block;
+    padding: 0;
+    position: relative;
+}
+
+.nav__link {
```
Comment (matine) [inline src/assets/scss/5-objects/_objects.nav.scss]: And @include remove-link-styles if it helps

File: src/assets/scss/6-components/_components.navbar.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
+// sass-lint:disable no-important, nesting-depth
```
Comment (matine) [inline src/assets/scss/6-components/_components.navbar.scss]: I would add the 'no-important' to that one line rather than the whole file

Comment (mikefrancis) [reply src/assets/scss/6-components/_components.navbar.scss]: I tried doing that and the linter was failing  😕

Comment (matine) [reply src/assets/scss/6-components/_components.navbar.scss]: sass-lint:disable-line no-important

File: src/assets/scss/1-settings/_settings.navbar.scss (hunk 2 above)
Comment (mikefrancis) [reply src/assets/scss/1-settings/_settings.navbar.scss]: I can't use `$icon-size-xxs` as the Sass compiler fails when it tries to do any maths with it

File: src/assets/scss/1-settings/_settings.navbar.scss [hunk 6]
```diff
@@ -0,0 +1,2 @@
+$dropdown-background-colour: $white;
+$caret-size: 10px;
```
Comment (matine) [inline src/assets/scss/1-settings/_settings.navbar.scss]: $icon-size-xxs?

File: src/templates/pages/navbar.hbs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        <!-- REQUIRED FILES -->
+        <section id="buttons-required">
+            <header class="content-block bg-light-grey">
+                <div class="container">
+                    <h3>Required files</h3>
+                </div>
+            </header>
+            <div class="content-block">
+                <div class="container">
+                    <p>SCSS files required: <code>_components.navbar.scss</code></p>
```
Comment (matine) [inline src/templates/pages/navbar.hbs]: Required files will now include the dropdown

File: src/assets/scss/1-settings/_settings.navbar.scss (hunk 6 above)
Comment (mikefrancis) [reply src/assets/scss/1-settings/_settings.navbar.scss]: Read my earlier comment

File: src/assets/scss/6-components/_components.navbar.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    > .nav__item {
+        &:hover > .nav__link {
+            background-color: $grey-very-light;
+        }
+    }
+}
+
+.dropdown > .nav__link {
```
Comment (matine) [inline src/assets/scss/6-components/_components.navbar.scss]: Does the linter not complain you haven't nested this?

Comment (mikefrancis) [reply src/assets/scss/6-components/_components.navbar.scss]: Nah, all good

File: src/assets/scss/1-settings/_settings.navbar.scss (hunk 6 above)
Comment (matine) [reply src/assets/scss/1-settings/_settings.navbar.scss]: I can't see one but I assume there's a good reason!

Comment (mikefrancis) [reply src/assets/scss/1-settings/_settings.navbar.scss]: > I can't use `$icon-size-xxs` as the Sass compiler fails when it tries to do any maths with it

**Pull request merged**